src/main/java/br/com/dev/eficiente/nosso/mercado/livre/controller/CategoryController.java

```diff
@@ -0,0 +1,33 @@
+package br.com.dev.eficiente.nosso.mercado.livre.controller;
+
+import br.com.dev.eficiente.nosso.mercado.livre.dto.Category;
+import br.com.dev.eficiente.nosso.mercado.livre.repository.CategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+public class CategoryController {
+
+    @Autowired
+    private CategoryRepository repository;
+
+    @PostMapping(value = "/category")
+    public ResponseEntity<String> addCategory(@RequestBody @Valid Category category){
```

Comment:
*Separamos as bordas externas do sistema do seu núcleo. Não ligamos parâmetros de requisição externa com objetos de domínio diretamente, assim como não serializamos objetos de domínio para respostas de API.*

Por mais que seja mais direto receber um objeto de domínio como argumento, os potenciais problemas são tão severos que eu deixei de lado a prática. 

* Qualquer setter que você colocar na Categoria pode ser utilizado por quem envia o parâmetro
* Alterações no construtor também são vistas automaticamente
* Alteração nos getters podem influenciar, caso a gente retornasse uma categoria

Basicamente você vincula o contrato do formulário ao objeto de domínio e deixa tudo muito frágil. Uma alteração no domínio, sem você perceber, impacta na borda externa. Sugiro a criação de um objeto que represente o formulário em si. 

*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Lembra que a validação já é algo com um fluxo dentro do Spring MVC. Sempre sugiro criar os validadores seguindo a interface deles ou usar criar um customizado da BeanValidation. Dessa forma você mantém o padrão de resposta de erro nos casos apropriados.